modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx

```diff
@@ -137,34 +222,227 @@ export default function EditObjectFolder({
 				/>
 			)}
 
-			{showModal.editObjectFolder && (
+			{modelBuilderModals.addObjectField && (
+				<ModalAddObjectField
+					baseResourceURL={baseResourceURL}
+					creationLanguageId={
+						selectedObjectDefinitionNode?.data
+							?.defaultLanguageId as Liferay.Language.Locale
+					}
+					objectDefinitionExternalReferenceCode={
+						selectedObjectDefinitionNodeERC
+					}
+					objectDefinitionName={
+						selectedObjectDefinitionNode?.data?.name as string
+					}
+					onAfterSubmit={(newObjectField) => {
+						const {edges, nodes} = store.getState();
+
+						if (selectedObjectDefinitionNode) {
```

Comment:
> Hey @Artartur, can you use selectedObjectDefinitionNode.data? as a modal rendering condition. So you don't need use as Liferay.Language.Locale, as string and others.

You can remove this `if` after the change above

Reply:
Done!